llvm/lib/Transforms/Scalar/StructurizeCFG.cpp

```diff
@@ -971,6 +971,10 @@ void StructurizeCFG::changeExit(RegionNode *Node, BasicBlock *NewExit,
     SubRegion->replaceExit(NewExit);
   } else {
     BasicBlock *BB = Node->getNodeAs<BasicBlock>();
+    for (BasicBlock *Succ : successors(BB)) {
+      if (Succ != NewExit)
+        Predicates[Succ].erase(BB);
+    }
```

Comment:
Probably should move this into killTerminator where the actual successor removal occurs

Reply:
does it mean that we indeed need to update `Predicates` here? I thought the no-update was on purpose.

Reply:
What makes you think it's on purpose? 

The predicates collects branch conditions used in the CFG, so it makes sense to me that you would need to adjust those when the CFG changes

Reply:
Hmm, alright. It has been sitting there for 10+ years so my first reaction is, probably it was designed in that way...